src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -95,6 +90,28 @@ bool AudioStreamAAudio::isSupported() {
     return openResult == 0;
 }
 
+// Static 'C' wrapper for the error callback method.
+// Launch a thread to handle the error.
+// That other thread can safely stop, close and delete the stream.
+void AudioStreamAAudio::internalErrorCallback(
+        AAudioStream *stream,
+        void *userData,
+        aaudio_result_t error) {
+    AudioStreamAAudio *oboeStream = reinterpret_cast<AudioStreamAAudio*>(userData);
+    if (oboeStream == nullptr) {
+        LOGE("%s() oboeStream is NULL!", __func__);
+    } else if (oboeStream->wasErrorCallbackCalled()) { // block double error callbacks
+        LOGE("%s() multiple error callbacks called!", __func__);
+    } else if (stream != oboeStream->getUnderlyingStream()) {
+        LOGD("%s() stream already closed", __func__);
```

Comment:
Can this ever happen (since the underlying stream is the only think that should make this callback)?

Reply:
This was what caused the assert in the old code. If it happens again then this will help us debug the problem.